### DISCUSSION TITLE
Generate Sections (Apps) with presets

### DISCUSSION THREAD

**Discussion by andiamine** [JoomCoder-com/JoomCCK#91] [Ideas]
I was thinking for another good features called Presets, where user can define a preset using json file, and generate Section from it quickly. is not like a wizard, but auto generate the app (section) from given json file. usefull to generate an App quickly. 

Any ideas and suggestions about this subject ?

**Comment by Giorgi625** [comment]
I have seen presets in template frameworks but do not know how we can use it in here, but if it will speed up our work this will be good. 
Maybe something like Zoo has? It has readymade presets for different occasions.
So it shows icons in admin dashboard like this and when you click you just enter name of the app and it creates. Of course it also shows settings so you can edit them or leave as default. 
![image](https://github.com/JoomCoder-com/JoomCCK/assets/16633699/2a0375a4-2611-4789-8312-594bebfa706e)

Of course even simillar thematic apps will have different fields, but there always will be fields which are used in any case. for example if I create movie catalog, I will have at least two types, Actors and Movies, maybe also Director, Tv-shows and so on, but that two are mandatory I guess and both types will have fields like for actors, gender, birthdate, nation, and so on, which I think everyone will need in their movie database so this preset should contain this kind of fields in advance which we think that every app about this would contain, and of course if there are more fields which user needs they can add them manually after app is created. 
Below is screenshot what Zoo creates when you select Cookbook app to be created:
![image](https://github.com/JoomCoder-com/JoomCCK/assets/16633699/ecc85990-d5d9-405d-8856-ba4dbf9beac3)

**Comment by Giorgi625** [comment]
JoomCCK is much more powerful then any other CCK extension on Joomla but some of them have features which are not yet in JoomCCK and noone can stop us by adding same features, in this way JoomCCK will have everything what others have and even more unique features which others do not.

**Reply by andiamine** [reply]
Yes my Joorney to Joomla dev begins with Mighty Resources, so still JoomCCK very advanced than any other CCK. needs just some good marketing and clean/easy UI/UX to attract more users.

**Comment by andiamine** [comment]
Thanks @Giorgi625, yes something similar to Zoo from Yootheme but much powerfull than that.

 This feature will have 4 main things:
- preset builder -> form with steps that let you build presets.
- preset generated as json files
- preset dependencies (for example if custom field not installed and required in preset)
- preset export/import

i think this feature will be different than packer and will replace it in future or we keep both if they have different use cases, we will see in future what will happen.
